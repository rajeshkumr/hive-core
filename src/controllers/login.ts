import { google, facebook } from "@lucia-auth/oauth/providers";
import auth from "../helpers/lucia";

export const googleAuth = google(auth, {
	clientId: process.env.GOOGLE_CLIENT_ID || "NOT_DEFINED", // env var
	clientSecret: process.env.GOOGLE_CLIENT_SECRET || "NOT_DEFINED",
  redirectUri: process.env.GOOGLE_CALLBACK || "NOT_DEFINED"
});

export const facebookAuth = facebook(auth, {
	clientId: process.env.FACEBOOK_CLIENT_ID || "NOT_DEFINED", // env var
	clientSecret: process.env.FACEBOOK_CLIENT_SECRET || "NOT_DEFINED",
  redirectUri: process.env.FACEBOOK_CALLBACK || "NOT_DEFINED"
});