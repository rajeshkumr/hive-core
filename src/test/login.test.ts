import { describe, expect, it, beforeEach, afterEach } from "bun:test";
import { Elysia } from "elysia";
import { googleAuth, facebookAuth } from "../controllers/login";

describe('Elysia', () => {

    beforeEach(() => {
        console.log("running test.");
    });

    afterEach(() => {
        console.log("done with test.");
    });

    it('return a response', async () => {
        facebookAuth.getAuthorizationUrl();
        googleAuth.getAuthorizationUrl();
        const serverMessage = "Elysia server running!";
        const app = new Elysia()
            .get('/', () => serverMessage)
        const response = await app.handle(
            new Request('http://localhost/')
        ).then(res => res.text())
        expect(response).toBe(serverMessage);
        expect(response).toMatchSnapshot();
    })
})