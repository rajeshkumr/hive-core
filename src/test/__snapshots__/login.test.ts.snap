// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`Elysia return a response 1`] = `"Elysia server running!"`;
